File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.java
class Solution {
    public int[] topKFrequent(int[] nums, int k) {

        // Hashmap to store the frequency of each number (k=number, v=frequency)
        HashMap<Integer, Integer> uniqueNumCount = new HashMap<>();

        // List of [frequence, number] pairs
        List<int[]> frequencyNumberPairs = new ArrayList<>();

        // Return Array of k most common numbers
        int[] mode = new int[k];

        // Fill the Hashmap with counts
        for (int num : nums) {
            if (!uniqueNumCount.containsKey(num)) {
                uniqueNumCount.put(num, 1);
            } else {
                int incrementedCount = uniqueNumCount.get(num) + 1;
                uniqueNumCount.replace(num, incrementedCount);
            }
        }

        for (Map.Entry<Integer, Integer> entry : uniqueNumCount.entrySet()) {
            frequencyNumberPairs.add(new int[] {entry.getValue(), entry.getKey()});
        }

        frequencyNumberPairs.sort((value, key) -> key[0] - value[0]);


        // Add values to the array by searching largest frequencies in list within map
        for (int i = 0; i < k; i++) {
            mode[i] = frequencyNumberPairs.get(i)[1];
        }


        return mode;
        
    }
}

File: Data Structures & Algorithms/two-integer-sum-ii/submission-1.cs
public class Solution {
    public int[] TwoSum(int[] numbers, int target) {

        int l = 0; int r = numbers.Length - 1;

        while (l < r) {
            int c = numbers[l] + numbers[r];
            if (c > target) {
                r--;
            } 
            else if (c < target) {
                l++;
            }
            else if (c == target) {
                return new int[] {l + 1, r + 1};
            }
        }
        return new int[0];
    }
}
